src/focusmanager.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global setTimeout, clearTimeout */
+
+import DOMEmitterMixin from '../ui/domemittermixin.js';
+import ObservableMixin from './observablemixin.js';
+import CKEditorError from './ckeditorerror.js';
+import mix from './mix.js';
+
+/**
+ * Observes if registered HTMLElements are focused. It is mainly used for keeping information if at least one of the editor component
+ * is currently focused which means that editor is focused.
+ *
+ * **Note** `focus` and `blur` listeners use event capturing, so it is only needed to register wrapper HTMLElement
+ * which contain other `focusable` elements. But note that this wrapper element has to be `focusable` too.
+ *
+ * @memberOf utils
+ * @mixes DOMEmitterMixin
+ * @mixes ObservableMixin
+ */
+export default class FocusManager {
```

Comment:
`FocusManager` is not the most fortunate name, but I can't find a better one... Maybe something like `FocusState` but I don't think that such odd names will be any better than the goold old `FocusManager`.